src/window.py

```diff
@@ -648,44 +670,65 @@ def hide_splanes(self):
             for s in self.splanes_actors:
                 s.VisibilityOff()
         else:
-            for s in self.splanes_actors:
-                s.VisibilityOn()
+            for i in range(len(self.splanes_actors)):
+                if not self.splanes_tree.topLevelItem(i).checkState(0) == Qt.CheckState.Checked:
+                    self.splanes_actors[i].VisibilityOn()
         self.reload_scene()
 
     def reload_splanes(self, splanes):
         self.hide_colorize()
         self._recreate_splanes(splanes)
-        self.splanes_list.clear()
+        self.splanes_tree.clear()
         for i in range(len(splanes)):
-            self.splanes_list.addItem("Figure" + " " + str(i + 1))
+            row = QTreeWidgetItem(self.splanes_tree)
+
+            if self.splanes_actors[i].GetVisibility():
+                row.setCheckState(0, Qt.CheckState.Unchecked)
+            else:
+                row.setCheckState(0, Qt.CheckState.Checked)
+
+            row.setText(1, str(i + 1))
+            row.setText(2, splanes[i].toFile())
             # self.splanes_list.addItem(self.locale.Plane + " " + str(i + 1))
 
         if len(splanes) > 0:
-            self.splanes_list.setCurrentRow(len(splanes) - 1)
+            self.splanes_tree.setCurrentItem(self.splanes_tree.topLevelItem(len(splanes) - 1))
         self.reload_scene()
 
     def _recreate_splanes(self, splanes):
         for p in self.splanes_actors:
             self.render.RemoveActor(p)
         self.splanes_actors = []
-        for p in splanes:
+        for i in range(len(splanes)):
```

Comment:
just for future, you can use _enumerate_

```python
for i, plane in enumerate(splanes):
```

this will create iterator with both index of object and object itself, construction will make code a bit simpler